### PR TITLE
Tweaks to the Iron Bank docker context

### PR DESCRIPTION
**Pull request by pugnascotia**
This PR makes a few tweaks to the Docker context that the ES build can generate, in order to align it more closely with what we submit to DSOP.

### CODE REVIEW

**Comment by pugnascotia** [comment]
@jmlrt Did you want the `validation` section omitted entirely for the ES artifact in `downloads.json`?

File: distribution/docker/src/docker/iron_bank/download.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "resources": [
     {
-      "url": "https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-${version}-linux-x86_64.tar.gz",
+      "url": "<artifact_path>/elasticsearch-${version}-linux-x86_64.tar.gz",
```
Comment (mark-vieira) [inline distribution/docker/src/docker/iron_bank/download.json:4]: What is this syntax? Does this value get replaced somewhere, and if so, how?

**Review by mark-vieira — APPROVED** [review]
One question, otherwise :+1:

File: distribution/docker/src/docker/iron_bank/download.json (hunk 1 above)
Comment (pugnascotia) [reply distribution/docker/src/docker/iron_bank/download.json:4]: It’s literally just a placeholder. Infra are planning a process to substitute the correct values during the unified release process.

File: distribution/docker/src/docker/Dockerfile [hunk 2]
```diff
@@ -364,6 +369,9 @@ LABEL name="Elasticsearch" \\
 RUN mkdir /licenses && \\
     cp LICENSE.txt /licenses/LICENSE
 <% } %>
+<% if (docker_base == 'iron_bank') { %>
+COPY LICENSE /licenses/LICENSE.addendum
```
Comment (jmlrt) [inline distribution/docker/src/docker/Dockerfile:373]: https://github.com/elastic/ironbank/blob/master/Elastic_License_w_gov_addendum_2020-10-08-1.txt contains standard license + addendum so it should completely replace the other license file.

Can you also add this license file to the generated build context tarball?

**Review by jmlrt — COMMENTED** [review]
>@jmlrt Did you want the validation section omitted entirely for the ES artifact in downloads.json?

Yes, the validation section should be omitted for ES artifacts.

File: distribution/docker/src/docker/Dockerfile (hunk 2 above)
Comment (pugnascotia) [reply distribution/docker/src/docker/Dockerfile:373]: The simplest thing to do would be to copy that file into the ES repo and add it into the generated build context. What do you think? Obviously having it in two places isn't ideal, but fetching the license file when the task run also feels fragile.

Comment (jmlrt) [reply distribution/docker/src/docker/Dockerfile:373]: I totally agree 👍

Comment (pugnascotia) [reply distribution/docker/src/docker/Dockerfile]: The license is now included in the repo and in the generated build context.

**Review by jmlrt — APPROVED** [review]
LGTM⛴

**Pull request merged**